jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -136,8 +136,151 @@ ecma_typedarray_create_object_with_typedarray (ecma_object_t *typedarray_p, /**<
   }
   else
   {
-    /** TODO: add other typed */
-    JERRY_UNREACHABLE ();
+    ecma_length_t byte_length = array_length << element_size_shift;
+    arraybuffer_p = ecma_arraybuffer_new_object (byte_length);
+    lit_utf8_byte_t *src_buf = ecma_arraybuffer_get_buffer (src_arraybuffer_p);
+    lit_utf8_byte_t *dst_buf = ecma_arraybuffer_get_buffer (arraybuffer_p);
+    uint8_t src_element_size_shift = ecma_typedarray_get_element_size_shift (typedarray_p);
+    double tmp;
+
+    for (uint32_t i = 0; i < array_length; i++)
+    {
+#define GET_VALUE_FROM_ARRAYBUFFER(type) \
+  type v = *((type *) src_buf); \
+  tmp = (double) v;
+
+#define SET_VALUE_TO_ARRAYBUFFER(type) \
+  int64_t v = (int64_t) tmp; \
+  *((type *) dst_buf) = (type) v;
+
+      /** check src type and get value from src typedarray */
+      switch (ecma_object_get_class_name (typedarray_p))
+      {
+        case LIT_MAGIC_STRING_INT8_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (int8_t);
```

Comment:
For me these macros are hard to read, since they return something but no return value.

    tmp = GET_VALUE_FROM_ARRAYBUFFER (int8_t, src_buf);

This would look better.